### PR TITLE
Refactor web sockets, improve exec, add attach.

### LINKED ISSUE #7: Support Pod Attach
Needed for Kubernetes "Gold" client status

### PR DESCRIPTION
**Pull request by brendandburns**
Works off of #24 

@arschles 

Fixes #7

### CODE REVIEW

File: exec-example.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+import { Exec } from './exec';
+
+
+let command = process.argv[2];
+
+let kc = new k8s.KubeConfig();
+kc.loadFromFile(process.env['HOME'] + '/.kube/config');
+
+let exec = new Exec(kc);
+exec.exec('default', 'nginx-4217019353-tmc2j', 'nginx', command, process.stdout, process.stderr, process.stdin, true /* tty */);
```
Comment (arschles) [inline exec-example.ts]: nit: can you remove the space at the end of the file and just keep the newline

File: exec.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import { KubeConfig } from '@kubernetes/typescript-node';
+
+export class Exec {
+    'handler': WebSocketHandler;
+
+    public constructor(config: KubeConfig) {
+        this.handler = new WebSocketHandler(config);
+    }        
+
+    // TODO: make command an array and support multiple args
```
Comment (arschles) [inline exec.ts]: nit: I've created https://github.com/kubernetes-client/typescript/issues/28 to track this

File: web-socket-handler.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    "channel.k8s.io"
+]
+
+export class WebSocketHandler {
+    'config': KubeConfig;
+
+    public static readonly StdinStream = 0;
+    public static readonly StdoutStream = 1;
+    public static readonly StderrStream = 2;
+    public static readonly StatusStream = 3;
```
Comment (arschles) [inline web-socket-handler.ts]: :+1:

**Review by arschles — APPROVED** [review]
A few nits, otherwise looks good.

Not blocking for this PR, since this library is relatively early, but as a follow-up we should add some tests for exec and attach

**Comment by arschles** [comment]
@brendandburns I meant to "request changes" on the review

File: exec-example.ts (hunk 1 above)
Comment (brendandburns) [reply exec-example.ts]: done (I think...)

**Comment by brendandburns** [comment]
Comments addressed, please re-check....

**Comment by brendandburns** [comment]
Something got messed up in the commit history, but the change looks right, I'm going to merge...

**Pull request merged**